sdk/metric/meter_test.go

```diff
@@ -2109,167 +2118,157 @@ func TestObservableDropAggregation(t *testing.T) {
-					"name":   "unregistered.observable.int64.counter",
-					"number": "int64",
+					name:   unregIntCntName,
+					number: "int64",
 				},
 				{
-					"error":  "observable instrument not registered for callback",
-					"msg":    "failed to record",
-					"name":   "unregistered.observable.float64.counter",
-					"number": "float64",
+					name:   unregFloatCntName,
+					number: "float64",
 				},
 			},
 		},
 		{
 			name: "drop int64 observable",
 			views: []View{
 				func(i Instrument) (Stream, bool) {
-					if strings.HasPrefix(i.Name, "observable.int64") {
+					if strings.HasPrefix(i.Name, intPrefix) {
 						return Stream{Aggregation: AggregationDrop{}}, true
 					}
 					return Stream{}, false
 				},
 			},
-			expectedObservableName: []string{
-				"observable.float64.counter",
-				"observable.float64.up.down.counter",
-				"observable.float64.gauge",
+			wantObservables: []string{
+				floatCntName, floatUDCntName, floatGaugeName,
 			},
-			expectedLogs: []map[string]interface{}{
+			wantUnregLogs: []log{
 				{
-					"error":  "observable instrument not registered for callback",
-					"msg":    "failed to record",
-					"name":   "unregistered.observable.int64.counter",
-					"number": "int64",
+					name:   unregIntCntName,
+					number: "int64",
 				},
 				{
-					"error":  "observable instrument not registered for callback",
-					"msg":    "failed to record",
-					"name":   "unregistered.observable.float64.counter",
-					"number": "float64",
+					name:   unregFloatCntName,
+					number: "float64",
 				},
 			},
 		},
 		{
 			name: "drop unregistered observable",
 			views: []View{
 				func(i Instrument) (Stream, bool) {
-					if strings.HasPrefix(i.Name, "unregistered.observable") {
+					if strings.HasPrefix(i.Name, unregPrefix) {
 						return Stream{Aggregation: AggregationDrop{}}, true
 					}
 					return Stream{}, false
 				},
 			},
-			expectedObservableName: []string{
-				"observable.int64.counter",
-				"observable.int64.up.down.counter",
-				"observable.int64.gauge",
-				"observable.float64.counter",
-				"observable.float64.up.down.counter",
-				"observable.float64.gauge",
+			wantObservables: []string{
+				intCntName, intUDCntName, intGaugeName,
+				floatCntName, floatUDCntName, floatGaugeName,
 			},
-			expectedLogs: []map[string]interface{}{},
+			wantUnregLogs: nil,
 		},
 	}
 	for _, tt := range testcases {
 		t.Run(tt.name, func(t *testing.T) {
-			logEntries := []map[string]interface{}{}
+			var unregLogs []log
 			global.SetLogger(
 				funcr.NewJSON(
 					func(obj string) {
 						var entry map[string]interface{}
 						_ = json.Unmarshal([]byte(obj), &entry)
-						logEntries = append(logEntries, entry)
+
+						// All unregistered observables should log `errUnregObserver` error.
+						// A observable with drop aggregation is also unregistered,
+						// however this is expected and should not log an error.
+						require.Equal(t, errUnregObserver.Error(), entry["error"])
```

Comment:
We can still continue the test to find other possible issues.

```suggestion
						assert.Equal(t, errUnregObserver.Error(), entry["error"])
```